src/games/strategy/triplea/ai/proAI/data/ProTerritoryManager.java

```diff
@@ -1,31 +1,1144 @@
+        if (!isCombatMove) {
+          potentialTerritories.add(myUnitTerritory);
+        }
+        for (final Territory potentialTerritory : potentialTerritories) {
+
+          // Find route over land checking whether unit can blitz
+          Route myRoute =
+              data.getMap().getRoute_IgnoreEnd(
+                  myUnitTerritory,
+                  potentialTerritory,
+                  ProMatches.territoryCanMoveLandUnitsThrough(player, data, myLandUnit, startTerritory, isCombatMove,
+                      enemyTerritories));
+          if (isCheckingEnemyAttacks) {
+            myRoute =
+                data.getMap().getRoute_IgnoreEnd(
+                    myUnitTerritory,
+                    potentialTerritory,
+                    ProMatches.territoryCanMoveLandUnitsThroughIgnoreEnemyUnits(player, data, myLandUnit,
+                        startTerritory, isCombatMove, enemyTerritories, clearedTerritories));
+          }
+          if (myRoute == null) {
+            continue;
+          }
+          if (myRoute.hasMoreThenOneStep()
+              && Match.someMatch(myRoute.getMiddleSteps(), Matches.isTerritoryEnemy(player, data))
+              && Matches.unitIsOfTypes(TerritoryEffectHelper.getUnitTypesThatLostBlitz(myRoute.getAllTerritories()))
+                  .match(myLandUnit)) {
+            continue; // If blitzing then make sure none of the territories cause blitz ability to be lost
+          }
+          final int myRouteLength = myRoute.numberOfSteps();
+          if (myRouteLength > range) {
+            continue;
+          }
+
+          // Add to route map
+          if (landRoutesMap.containsKey(potentialTerritory)) {
+            landRoutesMap.get(potentialTerritory).add(myUnitTerritory);
+          } else {
+            final Set<Territory> territories = new HashSet<Territory>();
+            territories.add(myUnitTerritory);
+            landRoutesMap.put(potentialTerritory, territories);
+          }
+
+          // Populate territories with land units
+          if (moveMap.containsKey(potentialTerritory)) {
+            moveMap.get(potentialTerritory).addMaxUnit(myLandUnit);
+          } else {
+            final ProTerritory moveTerritoryData = new ProTerritory(potentialTerritory);
+            moveTerritoryData.addMaxUnit(myLandUnit);
+            moveMap.put(potentialTerritory, moveTerritoryData);
+          }
+
+          // Populate unit move options map
+          if (unitMoveMap.containsKey(myLandUnit)) {
+            unitMoveMap.get(myLandUnit).add(potentialTerritory);
+          } else {
+            final Set<Territory> unitMoveTerritories = new HashSet<Territory>();
+            unitMoveTerritories.add(potentialTerritory);
+            unitMoveMap.put(myLandUnit, unitMoveTerritories);
+          }
+        }
+      }
+    }
+  }
+
+  private void findAirMoveOptions(final PlayerID player, final List<Territory> myUnitTerritories,
```

Comment:
Sort of both actually :)

These find*MoveOptions methods are really messy but also complex and difficult to refactor. I've essentially decided to just push them down as far as possible so that only TerritoryManager deals with them.